src/transformers/models/xlm/modeling_xlm.py

```diff
@@ -448,6 +446,9 @@ def __init__(self, config):
 
         # Initialize weights and apply final processing
         self.post_init()
+
+        if config.sinusoidal_embeddings:
+            create_sinusoidal_embeddings(config.max_position_embeddings, self.dim, out=self.position_embeddings.weight)
```

Comment:
Since it's a `XLMPreTrainedModel` we should also do it in the `_init_weights` no?  You can do "if isinstance(module, XLMModel):`